proto/pulumi/provider.proto

```diff
@@ -25,6 +25,22 @@ option go_package = "github.com/pulumi/pulumi/sdk/v3/proto/go;pulumirpc";
 // ResourceProvider is a service that understands how to create, read, update, or delete resources for types defined
 // within a single package.  It is driven by the overall planning engine in response to resource diffs.
 service ResourceProvider {
+
+    // Parameterize takes either a string array of command line inputs or a value embedded from sdk generation.
+    //
+    // Providers can be parameterized with either multiple extension packages (which don't define their own provider
+    // resources), or with a replacement package (which does define its own provider resource).
```

Comment:
How does the engine know if a provider is being parameterized by an extension package or a replacement provider? As I understand it, the engine will need to treat extended and replaced providers very differently.

Reply:
Yeh it will be part of the register resource call.